docs/data-companion/intro.md

```diff
@@ -0,0 +1,22 @@
+---
+order: 1
+parent:
+    title: Introduction
+    order: 1
+---
+
+# Introduction
+
+In [ADR-101](https://github.com/cometbft/cometbft/blob/thane/adr-084-data-companion-pull-api/docs/architecture/adr-101-data-companion-pull-api.md),
+it was proposed that a Data Companion be introduced as a solution to enable external applications to have an impact on the
+pruning of data on a node. The Data Companion works by leveraging the newly introduced gRPC APIs to control the [pruning](./pruning.md)
```

Comment:
The ADR talks about the API, the data companion is then an arbitrary implementation using this API for pruning so I would rephrase this,

Reply:
rewrite this part